mne/bem.py

```diff
@@ -6,14 +6,457 @@
 
 import numpy as np
 from scipy import linalg
+from os import path as op
 
 from .fixes import partial
-from .utils import verbose, logger
+from .utils import verbose, logger, deprecated, get_subjects_dir
 from .io.constants import FIFF
+from .io.write import (start_file, start_block, write_float, write_int,
+                       write_float_matrix, write_int_matrix, end_block,
+                       end_file)
+from .io.tag import find_tag
+from .io.tree import dir_tree_find
+from .io.open import fiff_open
 from .externals.six import string_types
 
 
 # ############################################################################
+# Compute BEM model
+
+# define VEC_DIFF(from,to,diff) {\
+# (diff)[X] = (to)[X] - (from)[X];\
+
+# The following approach is based on:
+#
+# de Munck JC: "A linear discretization of the volume conductor boundary
+# integral equation using analytically integrated elements",
+# IEEE Trans Biomed Eng. 1992 39(9) : 986 - 990
+#
+
+
+def _calc_beta(rk, rk1):
+    rkk1 = rk1 - rk
+    size = np.sqrt((rkk1 * rkk1).sum())
+    res = np.log((np.sqrt((rk * rk).sum())*size + np.dot(rk, rkk1)) /
+                 (np.sqrt((rk1 * rk1).sum())*size + np.dot(rk1, rkk1))) / size
+    return res
+
+
+def _lin_pot_coeff(fro, to_tri, to_rr):
+    """The linear potential matrix element computations"""
+    beta = np.empty(3)
+    bbeta = np.empty(3)
+    vec_omega = np.empty(3)
+    solid_eps = 4.0*np.pi / 1.0e6
+
+    y1 = to_rr[to_tri[0]] - fro
+    y2 = to_rr[to_tri[1]] - fro
+    y3 = to_rr[to_tri[2]] - fro
+    # This circularity makes things easy for us...
+    yy = [y1, y2, y3, y1, y3]
+    cross = np.cross(y1, y2)
+    triple = np.dot(cross, y3)
+
+    # XXX refactor with _get_solids?
+    l1 = np.sqrt((y1 * y1).sum())
```

Comment:
yeah, the reason I have it in this form is that I am almost certainly going to vectorize the operation, in which case having it in the `sum` form will be more convenient to start translating to higher dimensions